### PR TITLE
PLATUI-1207: Adding hmrcStandardLayout template

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        userResearchBanner = userResearchBannerUrl.map(url => UserResearchBanner(url = url)),
+        phaseBanner = phaseBanner,
+        displayHmrcBanner = displayHmrcBanner
+    )
+}
+
+@layout = {
+    @govukLayout(
+        pageTitle = pageTitle,
+        headBlock = if(head.isDefined) head else Some(hmrcHead(nonce = nonce)),
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:50]:
```suggestion
        headBlock = Some(hmrcHead(nonce = nonce, headBlock = headBlock)),
```

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        headerBlock = Some(headerBlock),
+        scriptsBlock = if(scripts.isDefined) scripts else Some(hmrcScripts(nonce = nonce)),
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:52]:
```suggestion
        scriptsBlock = Some(hmrcScripts(nonce = nonce, scriptsBlock = scriptsBlock)),
```

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+@(
+        serviceName: Option[String] = None,
+        pageTitle: Option[String] = None,
+        isWelshTranslationAvailable: Boolean = false,
+        signOutUrl: Option[String] = None,
+        homePageUrl: Option[String] = None,
+        userResearchBannerUrl: Option[String] = None,
+        displayHmrcBanner: Boolean = false,
+        phaseBanner: Option[PhaseBanner] = None,
+        head: Option[Html] = None,
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:30]:
```suggestion
        headBlock: Option[Html] = None,
```

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        scripts: Option[Html] = None,
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:31]:
```suggestion
        scriptsBlock: Option[Html] = None,
```

File: src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+      val updatedList = if (func(nextElement)) matchedElements :+ nextElement else matchedElements
+      filterElements(elements.nextAll(), func, updatedList)
+    }
+
+  implicit val fakeRequest = FakeRequest("GET", "/foo")
+
+  def buildApp(properties: Map[String, String] = Map.empty): Application =
+    new GuiceApplicationBuilder()
+      .configure(Map("play.i18n.langs" -> List("en", "cy")) ++ properties)
+      .build()
```
Comment (oscarduignan) [inline src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala:58]: @JoPintoPaul what do you think about using [GuiceOneAppPerSuite](https://www.playframework.com/documentation/2.7.0/api/scala/org/scalatestplus/play/guice/GuiceOneAppPerSuite.html) in this test? Not essential!

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        headBlock = Some(hmrcHead(nonce = nonce, headBlock = additionalHeadBlock)),
+        headerBlock = Some(headerBlock),
+        scriptsBlock = Some(hmrcScripts(nonce = nonce, scriptsBlock = additionalScriptsBlock)),
+        beforeContentBlock = if(isWelshTranslationAvailable) Some(hmrcLanguageSelectHelper()) else None,
+        footerBlock = Some(hmrcStandardFooter()),
+        mainContentLayout = mainContentLayout
+    )(contentBlock)
+}
+
+@layout
```
Comment (oscarduignan) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:59]: @JoPintoPaul should we unwrap this variable?

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        hmrcScripts: HmrcScripts,
+        defaultMainContent: TwoThirdsMainContent
+)
+
+@(
+        serviceName: Option[String] = None,
+        pageTitle: Option[String] = None,
+        isWelshTranslationAvailable: Boolean = false,
+        signOutUrl: Option[String] = None,
+        homePageUrl: Option[String] = None,
```
Comment (matthewmascord) [inline src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html:26]: Should we call this serviceUrl for consistency with hmrcStandardHeader? In the [Header](https://github.com/hmrc/play-frontend-govuk/blob/master/src/main/scala/uk/gov/hmrc/govukfrontend/views/viewmodels/header/Header.scala) viewmodel homePageUrl is used for https://www.gov.uk

File: src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala (hunk 5 above)
Comment (matthewmascord) [reply src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala:58]: 👍 This is building the app on each test.

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html (hunk 6 above)
Comment (JoPintoPaul) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html]: @oscarduignan Good suggestion, done.

File: src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html (hunk 7 above)
Comment (JoPintoPaul) [reply src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayout.scala.html]: @matthewmascord Good call, updated.

File: src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala (hunk 5 above)
Comment (JoPintoPaul) [reply src/test/scala/uk/gov/hmrc/hmrcfrontend/views/helpers/hmrcLayoutSpec.scala]: Corrected.

**Review by oscarduignan — APPROVED** [review]
looks good to me!

**Review by matthewmascord — APPROVED** [review]

**Review by oscarduignan — APPROVED** [review]

**Pull request merged**